api-call.py

```diff
@@ -1,29 +1,522 @@
+            print(f'{i}. {option}')
+
+    def get_user_choice(self) -> str:
+        """Get user input from menu
+
+        :returns: The user choice
+        """
+        return input('Enter your choice (0 or 1 or 2 or 3 or 4): ')
+
+    def get_url(self, choice: str) -> Dict[str, str]:
+        """Get URL for API
+
+        :param choice: user choice input
+        :returns: API URL Object after user choice
+        """
+
+        access_token = self.get_access_token()
+        if 'Bearer' not in access_token:
+            print(
+                'Error: Invalid access_token '
+                'of the application, could not connect'
+            )
+            exit()
+        header = {
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'Authorization': access_token
         }
+        url_obj = {}
+        if choice == '0':
+            return 'Exit'
+        elif choice == '1':
+            url_obj = {
+                'url': self.config['api_urls']['beaver_bus'],
+                'parameters': {},
+                'header': header
+            }
+        elif choice == '2':
+            url_obj = {
+                'url': self.config['api_urls']['terms'],
+                'parameters': {},
+                'header': header
+            }
+        elif choice == '3':
+            user_date = input('Enter Date (yyyy-mm-dd): ')
+            url_obj = {
+                'url_terms': self.config['api_urls']['terms'],
+                'parameters': {'date': user_date},
+                'header': header
+            }
+        elif choice == '4':
+            route_id = input('Enter Route Number: ')
+            url_obj = {
+                'url_routes': self.config['api_urls']['routes'],
+                'url_arrivals': self.config['api_urls']['arrivals'],
+                'url_vehicles': self.config['api_urls']['vehicles'],
+                'route_id': route_id,  'parameters': {},
+                'header': header
+            }
+        else:
+            print('Invalid choice. Please try again.')
+
+        return url_obj
+
+    def get_text_books_with_term_date(self, url_obj: Dict[str, str]) -> None:
+        """Get text books based on specified term and date
+        """
+
+        terms_data_response = self.get_api_data(
+            url_obj['url_terms'],
+            url_obj['parameters'],
+            url_obj['header']
+        )
+        if terms_data_response:
+            terms_data = terms_data_response.get('data', [])
+            if terms_data:
+                first_term = terms_data[0]
+                if 'attributes' in first_term:
+                    attributes = first_term['attributes']
+                    calendar_year = attributes.get('calendarYear')
+                    season = attributes.get('season')
+                    if not calendar_year:
+                        print("Error: 'calendarYear' attribute "
+                              'is missing or empty')
```

Comment:
Looks like the indentation here is still erroring. I think the correct formatting is:

```python
print(
    "Error: 'calendarYear' attribute "
    'is missing or empty'
)
```

This error appears a few other times below.

Reply:
fixed.